utils/lib/Alert.ts

```diff
@@ -81,6 +140,91 @@ class Alert extends Component<QuickstartConfigAlert[], QuickstartAlertInput[]> {
       };
     });
   }
+
+  /**
+   * Static method that gets the alert policy associated with a quickstart and it's current data sources
+   * @returns - object with alert policy ids, required data sources and NG errors
+   */
+  static async getAlertPolicyRequiredDataSources(quickstart: {
+    name: string;
+    dataSourceIds: string[];
+  }): Promise<
+    | { alertPolicy: AlertPolicyDataSource }
+    | { alertPolicy: null; errors: ErrorOrNerdGraphError[] }
+  > {
+    const { data, errors } = await fetchNRGraphqlResults<
+      AlertPolicyRequiredDataSourcesQueryVariables,
+      AlertPolicyRequiredDataSourcesQueryResults
+    >({
+      queryString: ALERT_POLICY_REQUIRED_DATA_SOURCES_QUERY,
+      variables: { query: `${quickstart.name} alert policy` },
+    });
+
+    const results = data?.actor?.nr1Catalog?.search?.results;
+
+    if (errors) {
+      return { alertPolicy: null, errors };
+    }
+
+    if (results === undefined || results.length === 0) {
+      const error = new Error(
+        `No alert policy for quickstart ${quickstart.name} exists`
+      );
+
+      return { alertPolicy: null, errors: [error] };
+    }
+
+    const hasFailed = quickstart.dataSourceIds.length > 1;
+
+    if (hasFailed) {
+      const error = new Error(
+        `Multiple Quickstart data sources detected for Quickstart: ${quickstart.name} with AlertPolicy: ${results[0].id} must update manually`
+      );
+      recordNerdGraphResponse(
+        hasFailed,
+        CUSTOM_EVENT.MULTIPLE_DATA_SOURCES_DETECTED
+      );
```

Comment:
Could we include a bit more information here? I'm just thinking we'll want to know the quickstart's name at least so that we can go find and update the alert policy

Reply:
I totally forgot about this, thanks for reminding me!